File: backend/IDE.Common/ModelsDTO/DTO/Project/ProjectInfoDTO.cs
using IDE.Common.DTO.Common;
using IDE.Common.Enums;
using System;

namespace IDE.Common.DTO.Project
{
    public class ProjectInfoDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public string ProjectLink { get; set; }
        public int CountOfSaveBuilds { get; set; }
        public int CountOfBuildAttempts { get; set; }
        public Language Language { get; set; }
        public ProjectType ProjectType { get; set; }
        public CompilerType CompilerType { get; set; }
        public AccessModifier AccessModifier { get; set; }
        public int AuthorId { get; set; }
        public string AuthorName { get; set; }
        public GitCredentialDTO GitCredential { get; set; }
    }
}

File: frontend/src/app/models/DTO/Project/projectInfoDTO.ts
/* Auto Generated */

import { Language } from "./../../Enums/language"
import { ProjectType } from "./../../Enums/projectType"
import { CompilerType } from "./../../Enums/compilerType"
import { AccessModifier } from "./../../Enums/accessModifier"
import { EditorSettingDTO } from "./../Common/editorSettingDTO"
import { GitCredentialDTO } from "./../Common/gitCredentialDTO"

export interface ProjectInfoDTO {
    id: number;
    name: string;
    description: string;
    createdAt: Date;
    projectLink: string;
    countOfSaveBuilds: number;
    countOfBuildAttempts: number;
    language: Language;
    projectType: ProjectType;
    compilerType: CompilerType;
    accessModifier: AccessModifier;
    editorProjectSettings: EditorSettingDTO;
    authorId: number;
    authorName: string;
    gitCredential: GitCredentialDTO;
    color: string;
    amountOfMembers: number;
}
